pkg/clusterapi/bottlerocket.go

```diff
@@ -120,23 +123,54 @@ func SetBottlerocketControlContainerImageInKubeadmConfigTemplate(kct *bootstrapv
 
 // SetBottlerocketHostConfigInKubeadmControlPlane sets bottlerocket specific kernel settings in kubeadmControlPlane.
 func SetBottlerocketHostConfigInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, hostOSConfig *anywherev1.HostOSConfiguration) {
-	if hostOSConfig == nil || hostOSConfig.BottlerocketConfiguration == nil || hostOSConfig.BottlerocketConfiguration.Kernel == nil ||
-		hostOSConfig.BottlerocketConfiguration.Kernel.SysctlSettings == nil {
+	if hostOSConfig == nil || hostOSConfig.BottlerocketConfiguration == nil || (hostOSConfig.BottlerocketConfiguration.Kernel == nil &&
```

Comment:
nit: can we do something like

```suggestion
	if hostOSConfig == nil || hostOSConfig.BottlerocketConfiguration == nil {
	return nil
	}
```

then update `hostConfig()` to include the boot settings, use it in all below, to remove the duplicate code